src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +201,46 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        // Only use gold under specific conditions that we know work
+
+        let gold_exists = match env::var_os("PATH") {
+            Some(ref env_path) => {
+                env::split_paths(env_path).any(|p| {
+                    let mut p = PathBuf::from(p);
+                    p.push("ld.gold");
+                    p.exists()
+                })
+            }
+            None => false
+        };
+        let host_is_linux = cfg!(target_os = "linux");
+        // Defensively prevent trying to use gold for bogus cross-targets.
+        let host_is_target =
+            self.sess.target.target.arch == env::consts::ARCH &&
```

Comment:
This ends up disabling golds in cross 32/64 builds, right? (could that be allowed?)